codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/ValidationGenerator.java

```diff
@@ -565,4 +438,147 @@ private StringBuilder addUTFCheck(
     }
     return UTFCheck;
   }
+
+  //dataSource in non-final; Not sure if we need it as mutable.
```

Comment:
Its non final because of statement `dataSource = "Value"`

Reply:
The convention in Java is to not modify an input parameter. I wasn't sure if we need it as mutable or not hence left is as such (and that's what comment explains).